CMakeLists.txt

```diff
@@ -69,6 +69,11 @@ set( DETRAY_BENCHMARKS_REP 1 CACHE STRING
 # Clean up.
 unset( DETRAY_BUILD_CUDA_DEFAULT )
 
+# Flag the setup of the io library
+if ( DETRAY_IO_CSV OR DETRAY_IO_JSON )
+   set( DETRAY_IO ON )
```

Comment:
I don't like this `DETRAY_IO` variable. :frowning: The `DETRAY_IO_CSV OR DETRAY_IO_JSON` expression is needed exactly once in the code. When we decide whether to use the `io/` subdirectory or not. So instead of introducing a while new variable, which you would then make use of in `io/CMakeLists.txt`, let's just change the inclusion of the `io/` subdirectory in this configuration file to happen like:

```cmake
if( DETRAY_IO_CSV OR DETRAY_IO_JSON )
   add_subdirectory( io )
endif()
```